### ISSUE TITLE
Overwrite allowed on Travis with open connection

### ISSUE THREAD

**Issue by bjcairns** [bjcairns/ukbschemas#5]
On Travis-CI, build 6 failed because the test suite failed with the following report:
```
  ── 1. Failure: create_schema_db() fails to overwrite when db is connected (@test
  `{ ... }` did not throw an error.
  
  ══ testthat results  ═══════════════════════════════════════════════════════════
  OK: 19 SKIPPED: 1 WARNINGS: 1 FAILED: 1
  1. Failure: create_schema_db() fails to overwrite when db is connected (@test-create-schema-db.R#73) 
```

The failing test passes on my local Win10 OS, so maybe this is an OS issue?

The current work-around is to skip the test except on Windows, but it would be nice to know whether this goes deeper, e.g. if on Linux systems `file.remove()` works even when there is an open database connection. If so, some more robust way to check for open connections would be nice to have.

**Comment by elreda** [comment]
It seems to be an OS thing indeed. On MacOS for example, `file.remove()` has no problem deleting a file with an open database connection. 

Based on the answer below on SO, running `lsof` on the file could be one way to "manually" check for open connections on Unix systems before trying to remove the file. 
https://stackoverflow.com/questions/12138260

**Comment by bjcairns** [comment]
Going to close this - planning to migrate to GitHub Actions (per #34) and will see where that leaves us.
